AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -141,29 +141,27 @@ + (BOOL)openURL:(NSURL *)url {
 }
 
 + (void)login {
-
-  // TODO protect with canBeUsed.
-  [[MSIdentity sharedInstance] login];
+  @synchronized(self) {
+    if ([[MSIdentity sharedInstance] canBeUsed]) {
+      [[MSIdentity sharedInstance] login];
+    }
+  }
 }
 
 - (void)login {
-  @synchronized(lock) {
-    if (self.clientApplication == nil && self.identityConfig == nil) {
-      self.loginDelayed = YES;
-      return;
-    }
-    self.loginDelayed = NO;
-    [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
-                                  completionBlock:^(MSALResult *result, NSError *e) {
-                                    // TODO: synchronize accessToken assignment if it has threading issues
-                                    if (e) {
-                                      MSLogError([MSIdentity logTag], @"Couldn't initialize authentication client. Error: %@", e);
-                                    } else {
-                                      NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
-                                      self.accessToken = result.accessToken;
-                                    }
-                                  }];
+  if (self.clientApplication == nil || self.identityConfig == nil) {
+    self.loginDelayed = YES;
+    return;
   }
+  self.loginDelayed = NO;
+  [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
+                                completionBlock:^(MSALResult *result, NSError *e) {
+                                  if (e) {
+                                    MSLogError([MSIdentity logTag], @"Couldn't initialize authentication client. Error: %@", e);
```

Comment:
"Couldn't initialize authentication client" error message is somewhat misleading. Initialization of MSAL client application is done at `(void)configAuthenticationClient`; while the acquire token method is logging user in.
Please improve this error message.

Reply:
Changed it to `User login failed` (like for Android). I hope it makes sense now. Thanks.